projects/example-app/src/app/auth/containers/login-page.component.ts

```diff
@@ -9,7 +9,7 @@ import { LoginPageActions } from '@example-app/auth/actions';
   template: `
     <bc-login-form
       (submitted)="onSubmit($event)"
-      [pending]="pending$ | async"
+      [pending]="(pending$ | async) === true"
```

Comment:
```suggestion
      [pending]="(pending$ | async)!"
```

Reply:
Do these have the same behavior?

Reply:
Yes, but to be consistent, let's use the same syntax for all state observables that are unwrapped via `async` pipe.

Suggestion from Angular documentation: https://angular.io/guide/template-typecheck#strict-null-checks

Reply:
Nice, that's a convenience I didn't know about.

Reply:
According to that doc, they do not have the same behavior:

> In the template, include the non-null assertion operator ! at the end of a nullable expression, such as <user-detail [user]="user!"></user-detail>.
In this example, the compiler disregards type incompatibilities in nullability, just as in TypeScript code.

It's nbd, but since this is the example app we're choosing to do assertions which may be incorrect instead of truly "strict" handling of `async`'s emissions of `null`.

Reply:
The doc further advises adjusting Inputs to components to say whether they can handle falsy values: 

> As a library author, you can take several measures to provide an optimal experience for your users. First, enabling strictNullChecks and including null in an input's type, as appropriate, communicates to your consumers whether they can provide a nullable value or not.

Reply:
> they do not have the same behavior

If we still talk about `pending$`, both solutions will work the same: `(pending$ | async)!` and `(pending$ | async) === true` and the return type of both expressions will be `boolean`. I suggested the first solution, to be consistent with the others.

You quoted sections that are not related to the `async` pipe :) I meant this part:

> The async pipe currently assumes that the Observable it subscribes to can be asynchronous, which means that it's possible that there is no value available yet. In that case, it still has to return something—which is null. In other words, **the return type of the async pipe includes null, which might result in errors in situations where the Observable is known to emit a non-nullable value synchronously.**

---

> It's nbd, but since this is the example app we're choosing to do assertions which may be incorrect instead of truly "strict" handling of async's emissions of null.

In case of `pending$` or `books$` for example, `async` will never return `null`.

---

IMO, the best will be to create view model selectors for all pages and avoid non-null assertions at all, but that could be an idea for another PR :)

Reply:
Observables have `null` added to their return types because they're observables, and there might not be a value yet.
This isn't the case for selectors, because there will always be an initial value, unless a `null` or `undefined` value is explicitly returned by the selector.
In these cases, I think there's always a value, hence the suggestion to add a bang operator to these async pipes.

Reply:
Thanks for the investigation on part of my knowledge gaps 😅

Reply:
Knowing this, I wonder if the ngrxLet directive in coordination with the return type of store.select could meet strictness and know that whatever the observable's value is, is guaranteed.

Reply:
Just thinking about that lib out loud, not saying it needs to be used here.

Reply:
> Thanks for the investigation on part of my knowledge gaps 😅

No worries David, thank you for your efforts to make NgRx codebase better! :)